packages/db/src/network/test/mockProvider.ts

```diff
@@ -35,22 +39,22 @@ export const mockProvider = (options: {
             : undefined;
 
           return callback(null, {
-            jsonrpc,
+            jsonrpc: jsonrpc as JsonRpcIdentifier,
             id,
-            result
+            result: result as any
           });
         }
         case "net_version": {
           const result = networkId;
           debug("result %o", result);
 
           return callback(null, {
-            jsonrpc,
+            jsonrpc: jsonrpc as JsonRpcIdentifier,
             id,
             result
           });
         }
       }
     }
-  };
+  } as Web3BaseProvider;
```

Comment:
Not sure if we should type cast this if it's being used elsewhere as a non web3.js provider